src/Microsoft.TemplateEngine.Utils/InMemoryFileSystem.cs

```diff
@@ -886,5 +886,18 @@ public void SetLastWriteTimeUtc(string file, DateTime lastWriteTimeUtc)
 
             targetFile.LastWriteTimeUtc = lastWriteTimeUtc;
         }
+
+        public IDisposable WatchFileChanges(string filepath, FileSystemEventHandler fileChanged)
+        {
+            return new MemoryStream();//Just some disposable dummy
```

Comment:
consider this instead
https://docs.microsoft.com/en-us/previous-versions/dotnet/reactive-extensions/hh229382(v=vs.103)

Reply:
what's the decision on this?  Re-inventing an existing library, is acquiring technical debt.  We will likely have to upgrade to Rx extensions later.

Reply:
I mentioned in PR chat that I decided to not adopt ReactiveExtensions, but after further thinking I now opened Issue, that we will discuss as part of Public API changes planned for April. https://github.com/dotnet/templating/issues/2933